ai-server/app/music/run_musicgen.py

```
@@ -1,30 +1,40 @@
+def sanitize_filename(prompt: str) -> str:
+    return re.sub(r'[\\/*?:"<>|]', "", prompt)[:30]
+
+
+def generate_music(prompt: str, duration: int = 10) -> dict:
     model.set_generation_params(duration=duration)
-    wav = model.generate([prompt], progress=False)
+
+    try:
+        wav = model.generate([prompt], progress=False)
+    except Exception as e:
+        raise GenerationError(f"Music generation failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)